addOns/scripts/src/test/java/org/zaproxy/zap/extension/scripts/scanrules/PassiveScriptScanRuleUnitTest.java

```diff
@@ -107,6 +108,24 @@ void shouldHandleNullReferences() throws Exception {
         assertThat(alert.getReference(), is(equalTo("")));
     }
 
+    @Test
+    void shouldHonourScriptEngineThreadedPropertyOnAppliesToHistoryType() throws Exception {
```

Comment:
Ideally we would test whether the `appliesToHistoryType` call was done in a synchronized block or not, but I don't suppose that's possible / straightforward.

Reply:
That's what I wanted to do :)